pymedphys/_dicom/delivery/core.py

```diff
@@ -134,42 +119,68 @@ def _from_dicom_beam(cls, beam, meterset):
         if meterset is None:
             raise ValueError("Meterset should not ever be None")
 
-        leaf_boundaries = beam.BeamLimitingDeviceSequence[-1].LeafPositionBoundaries
+        beam_limiting_device_sequence = beam.BeamLimitingDeviceSequence
+
+        rt_beam_limiting_device_types = {
+            item.RTBeamLimitingDeviceType for item in beam_limiting_device_sequence
+        }
+
+        if rt_beam_limiting_device_types != set(["MLCX", "ASYMY"]):
+            raise ValueError(
+                "Currently only DICOM files that contain "
+                "exactly MLCX and ASYMY are supported. "
+                f"{rt_beam_limiting_device_types} is not supported."
+            )
+
+        mlc_sequence = [
+            item
+            for item in beam_limiting_device_sequence
+            if item.RTBeamLimitingDeviceType == "MLCX"
+        ]
+
+        if len(mlc_sequence) != 1:
+            raise ValueError("Expected there to be only one device labelled as MLCX")
+
+        mlc_limiting_device = mlc_sequence[0]
+
+        leaf_boundaries = mlc_limiting_device.LeafPositionBoundaries
         leaf_widths = np.diff(leaf_boundaries)
 
-        assert beam.BeamLimitingDeviceSequence[-1].NumberOfLeafJawPairs == len(
-            leaf_widths
-        )
+        if mlc_limiting_device.NumberOfLeafJawPairs != len(leaf_widths):
+            raise ValueError(
+                "Expected number of leaf pairs to be the same as "
+                "the length of leaf widths"
+            )
+
         num_leaves = len(leaf_widths)
 
         control_points = beam.ControlPointSequence
 
-        beam_limiting_device_position_sequences = get_cp_attribute_leaning_on_prior(
+        beam_limiting_device_position_sequence = rtplan.get_cp_attribute_leaning_on_prior(
             control_points, "BeamLimitingDevicePositionSequence"
         )
 
-        mlcs = [
-            sequence[-1].LeafJawPositions
-            for sequence in beam_limiting_device_position_sequences
-        ]
```

Comment:
I am annoyed with past Simon that he chose to reference the "MLC" sequence entry by its index, not by its accompanying "MLCX" type. I can't believe I did that... Not sure what I was thinking...